sql/core/src/test/scala/org/apache/spark/sql/MiscFunctionsSuite.scala

```diff
@@ -285,6 +285,48 @@ class MiscFunctionsSuite extends QueryTest with SharedSparkSession {
     assert(df.selectExpr("random(1)").collect() != null)
     assert(df.select(random(lit(1))).collect() != null)
   }
+
+  test("SPARK-49306 nullifzero and zeroifnull functions") {
```

Comment:
This test suite is for misc functions mostly. Could you move expression tests to `ConditionalExpressionSuite` and end-to-end tests to `conditional-functions.sql`, please.

Reply:
Sure, it turns out all these were end to end tests so I just moved them all to `conditional-functions.sql`.